packages/core/parcel-bundler/src/Bundler.js

```diff
@@ -566,6 +570,14 @@ class Bundler extends EventEmitter {
 
     // Call the delegate to get implicit dependencies
     let dependencies = processed.dependencies;
+
+    // exlucde externals
+    if (this.externals.length && !this.options.hmr) {
```

Comment:
```suggestion
    if (this.externals && this.externals.length && !this.options.hmr) {
```

To fix a test: `Cannot read property 'length' of undefined`